services/workflows-service/src/transaction/transaction.controller.external.ts

```diff
@@ -333,42 +336,82 @@ export class TransactionControllerExternal {
     @Query() getTransactionsByAlertParameters: GetTransactionsByAlertDto,
     @CurrentProject() projectId: types.TProjectId,
   ) {
-    const alert = await this.prisma.alert.findUnique({
-      where: {
-        id: getTransactionsByAlertParameters.alertId,
+    const queryArgs = this.scopeService.scopeFindOne(
+      {
+        include: {
+          alertDefinition: true,
+        },
+        where: {
+          id: getTransactionsByAlertParameters.alertId,
+        },
       },
-    });
+      [projectId],
+    );
+
+    const alert = await this.prisma.alert.findUnique(queryArgs);
 
     if (!alert) {
       throw new errors.NotFoundException(
         `Alert with id ${getTransactionsByAlertParameters.alertId} not found`,
       );
     }
 
-    const alertCreatedAt = alert.createdAt;
-
-    const alertDefinition = await this.prisma.alertDefinition.findUnique({
-      where: {
-        id: alert.alertDefinitionId,
-      },
-    });
+    let filters: GetTransactionsByAlertDto = {
+      ...getTransactionsByAlertParameters,
+    };
 
-    if (!alertDefinition) {
+    if (!alert.alertDefinition) {
       throw new errors.NotFoundException(`Alert definition not found for alert ${alert.id}`);
     }
+    const inlineRule = alert.alertDefinition.inlineRule as InlineRule;
 
-    const inlineRule = alertDefinition.inlineRule as InlineRule;
-    // @ts-expect-error
-    const { timeValue, timeUnit } = inlineRule.options;
+    if (!inlineRule || !inlineRule.options) {
+      return filters;
+    }
+    if (!filters.endDate) {
+      filters.endDate = alert.updatedAt;
+    } else if (
+      !filters.startDate &&
```

Comment:
we can remove start date, this endpoint is get transaction by alert - so we can remove almost all filters and let the backend handle what he fetches by the alert definition, no?

we have the other endpoint if the user want to fetch by other filters like end date etc..